learningequality/studio
# Pagination integration proof of concept

## Summary
### Description of the change(s) you made

After numerous issues arising because of pagination, this creates a prototype to ensure full integration of both the ValuesViewset and the frontend resource layer with both ordering and pagination.

This means that we can use standard Django Rest Framework `ordering` syntax for specifying the ordering of results from the API (https://www.django-rest-framework.org/api-guide/filtering/#orderingfilter) and this will be properly interpreted by both the backend and the IndexedDB `where` wrapper command.

In addition, this adds support in the backend for PageNumberPagination to be added to any ValuesViewset class when needed, and to leverage the efficiency gains that were added specifically to the AdminChannel and AdminUser viewsets.

In the frontend it adds support for PageNumberPagination and LimitOffsetPagination for IndexedDB queries through `where`.


### Manual verification steps performed
Checked all places that ordering and pagination are currently used:
* Admin channel and user viewsets
* Content node search viewsets
* Catalog listing for a logged in user (changed to using `where` and verified that pagination worked as expected)

### Screenshots (if applicable)


### Does this introduce any tech-debt items?

___
## Reviewer guidance
### How can a reviewer test these changes?



### Are there any risky areas that deserve extra testing?



## References


## Comments


----

### Contributor's Checklist


PR process:


Studio-specifc:



Testing:

___

### Reviewer's Checklist
#### This section is for reviewers to fill out.
